Directory.Packages.props

```diff
@@ -84,37 +84,37 @@
   -->
   <ItemGroup>
     <PackageVersion Include="BenchmarkDotNet" Version="0.15.2" />
-    <PackageVersion Include="CommandLineParser" Version="[2.9.1,3.0)" />
+    <PackageVersion Include="CommandLineParser" Version="2.9.1" />
     <PackageVersion Include="GitHubActionsTestLogger" Version="2.4.1" />
-    <PackageVersion Include="Grpc.AspNetCore" Version="[2.59.0,3.0)" />
-    <PackageVersion Include="Grpc.AspNetCore.Server" Version="[2.59.0, 3.0)" />
-    <PackageVersion Include="Grpc.Tools" Version="[2.59.0,3.0)" />
-    <PackageVersion Include="Google.Protobuf" Version="[3.22.5,4.0)" />
-    <PackageVersion Include="Grpc" Version="[2.44.0,3.0)" />
-    <PackageVersion Include="Grpc.Net.Client" Version="[2.52.0,3.0)" />
+    <PackageVersion Include="Grpc.AspNetCore" Version="2.59.0" />
```

Comment:
I see we're removing upper version constraints across multiple packages. Could you please explain why this is needed.

Reply:
So we get pull requests to update them and can then review them.

These are all test dependencies so there shouldn't be any harm in keeping them up to date.

There's more detailed commentary on the changes being made in the linked PR that was originally opened.